cmake/FindCompiler.cmake

```diff
@@ -38,8 +38,13 @@ if(is_dpcpp)
         -fsycl-targets=nvptx64-nvidia-cuda -fsycl-unnamed-lambda)
       list(APPEND UNIX_INTERFACE_LINK_OPTIONS
         -fsycl-targets=nvptx64-nvidia-cuda)
+    elif(ENABLE_ROCBLAS_BACKEND)
```

Comment:
I don't think `elif` is the right syntax for CMake. `elseif` is what you are looking for. https://cmake.org/cmake/help/v3.15/command/if.html